packages/shared/src/utils/contract/contract.tsx

```diff
@@ -344,3 +345,6 @@ export const getSortedTradeTypes = (array: string[] = []) => {
     }
     return array;
 };
+
+export const isForwardStartingBuyTransaction = (transactionType: string, shortcode: string, transactionTime: number) =>
+    transactionType === 'buy' && !!isForwardStarting(shortcode, transactionTime);
```

Comment:
please cover this new function with tests🙏🏻

Reply:
Will cover